Tests/SotoTests/Services/ApiGatewayV2/APIGatewayV2Tests.swift

```diff
@@ -47,22 +47,26 @@ class APIGatewayV2Tests: XCTestCase {
         }
         /// If we create a rest api for each test, when we delete them APIGateway will throttle
         /// and we will most likely not delete the all APIs so we create one API to be used by all tests
-        XCTAssertNoThrow(try runThrowingTask(on: Self.client.eventLoopGroup.any()) {
-            do {
-                Self.restApiId = try await self.createRestApi(name: self.restApiName)
-            } catch {
-                print("Failed to create APIGateway rest api, error: \(error)")
-                throw error
+        Task {
+            await XCTAsyncAssertNoThrow {
+                do {
+                    Self.restApiId = try await self.createRestApi(name: self.restApiName)
+                } catch {
+                    print("Failed to create APIGateway rest api, error: \(error)")
+                    throw error
+                }
             }
-        })
+        }.syncAwait()
     }
 
     override class func tearDown() {
```

Comment:
Why not use the async teardown and setup functions to avoid needing to putting it in a task and awaiting on it?

Reply:
Ah I missed that - yeah that's a massive API hole